crates/bevy_scene/src/dynamic_scene.rs

```diff
@@ -76,17 +78,79 @@ impl DynamicScene {
         scene
     }
 
+    /// Create a new dynamic scene from a given world at a given root.
+    pub fn from_world_at_root(
+        world: &World,
+        root: Entity,
+        type_registry: &AppTypeRegistry,
+    ) -> Self {
+        let mut scene = DynamicScene::default();
+        let type_registry = type_registry.read();
+
+        // Extract all entities that are a descendent of root
+        let mut entities_of_interest = Vec::new();
+        let mut entities_to_process = vec![root];
+        while let Some(entity) = entities_to_process.pop() {
```

Comment:
This recursive "get me all entities in this tree" method feels like it should be a standalone method over in `bevy_hierarchy`. That will make it more robust to refactors, and expose a useful API to end users.